ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java

```diff
@@ -125,73 +141,52 @@ public PersistentIndex(String datadir, ScheduledExecutorService scheduler, Log l
    * @param time the time instance to use
    * @throws StoreException
    */
-  protected PersistentIndex(String datadir, ScheduledExecutorService scheduler, Log log, StoreConfig config,
+  PersistentIndex(String datadir, ScheduledExecutorService scheduler, Log log, StoreConfig config,
       StoreKeyFactory factory, MessageStoreRecovery recovery, MessageStoreHardDelete hardDelete, StoreMetrics metrics,
       Journal journal, Time time) throws StoreException {
+    File[] indexFiles = new File(datadir).listFiles(INDEX_FILE_FILTER);
+    if (indexFiles == null) {
+      throw new StoreException("Could not read index files from directory [" + datadir + "]",
+          StoreErrorCodes.Index_Creation_Failure);
+    }
     try {
+      this.dataDir = datadir;
+      this.log = log;
       this.time = time;
       this.scheduler = scheduler;
       this.metrics = metrics;
-      logSegment = log.getFirstSegment();
-      File indexDir = new File(datadir);
-      File[] indexFiles = indexDir.listFiles(new IndexFilter());
       this.factory = factory;
       this.config = config;
-      persistor = new IndexPersistor();
-      hardDeleter = new HardDeleter(config, metrics, datadir, log, this, hardDelete, factory, time);
       this.hardDelete = hardDelete;
       this.journal = journal;
-      Arrays.sort(indexFiles, new Comparator<File>() {
-        @Override
-        public int compare(File o1, File o2) {
-          if (o1 == null || o2 == null) {
-            throw new NullPointerException("arguments to compare two files is null");
-          }
-          // File name pattern for index is offset_name. We extract the offset from
-          // name to compare
-          int o1Index = o1.getName().indexOf("_", 0);
-          long o1Offset = Long.parseLong(o1.getName().substring(0, o1Index));
-          int o2Index = o2.getName().indexOf("_", 0);
-          long o2Offset = Long.parseLong(o2.getName().substring(0, o2Index));
-          if (o1Offset == o2Offset) {
-            return 0;
-          } else if (o1Offset < o2Offset) {
-            return -1;
-          } else {
-            return 1;
-          }
-        }
-      });
+      this.maxInMemoryIndexSizeInBytes = config.storeIndexMaxMemorySizeBytes;
+      this.maxInMemoryNumElements = config.storeIndexMaxNumberOfInmemElements;
 
+      logAbsoluteZeroOffset = new Offset(log.getFirstSegment().getName(), 0);
```

Comment:
I know you might have thought about this, but may I know the reason to choose 0 as startOffset and not the startOffset of the first segment. Lets say not writes have gone into the store and if someone tries to find the totalBytesRead, wouldn't we return startOffset - 0, rather than 0. Because in find the total bytes read as part of findEntriesSince(), we calculate the diff from 0 and not startOffset of the first segment. Or a I missing something.